src/Algolia/ManualIndexer.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace AppBundle\Algolia;
+
+use Algolia\AlgoliaSearchBundle\Indexer\Indexer;
+use Doctrine\ORM\EntityManagerInterface;
+
+class ManualIndexer implements ManualIndexerInterface
+{
+    /**
+     * @var \Algolia\AlgoliaSearchBundle\Indexer\ManualIndexer
+     */
+    private $indexer;
+
+    public function __construct(Indexer $algolia, EntityManagerInterface $manager)
+    {
+        $this->indexer = $algolia->getManualIndexer($manager);
```

Comment:
Two possibilities:
- use a factory method for the service declaration and type hint `ManualIndexer` directly in the constructor
- keep both references and build the service only when `index()` is called

In this case I guess that if the service is construct the method will be called so the first one should be the way to go, not sure though.